launcher/src/json-utils.ts

```diff
@@ -0,0 +1,65 @@
+/**********************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+/**
+ * Recursively copies all properties from the first object to the next.
+ * If the property exists in the next object, its value will be updated.
+ */
+export function mergeFirstWithSecond(first: any, second: any) {
```

Comment:
I would name it just `merge()`
It's clear enough that the method merges given objects